service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/handlers/C2UploadedEventHandlerEmailTemplateTest.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.hmcts.reform.fpl.handlers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import uk.gov.hmcts.reform.fpl.events.C2UploadedEvent;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.C2DocumentBundle;
+import uk.gov.hmcts.reform.fpl.model.notify.c2uploaded.C2UploadedTemplate;
+import uk.gov.hmcts.reform.fpl.request.RequestData;
+import uk.gov.hmcts.reform.fpl.service.email.NotificationService;
+import uk.gov.hmcts.reform.fpl.service.email.content.C2UploadedEmailContentProvider;
+import uk.gov.hmcts.reform.fpl.testingsupport.email.EmailTemplateTest;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+import static org.mockito.BDDMockito.given;
+
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.enums.UserRole.LOCAL_AUTHORITY;
+import static uk.gov.hmcts.reform.fpl.handlers.NotificationEventHandlerTestData.AUTH_TOKEN;
+import static uk.gov.hmcts.reform.fpl.testingsupport.email.EmailContent.emailContent;
+import static uk.gov.hmcts.reform.fpl.testingsupport.email.SendEmailResponseAssert.assertThat;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.caseData;
+
+@SpringBootTest(classes = {
+    C2UploadedEventHandler.class,
+    NotificationService.class,
+    ObjectMapper.class
+})
+
+public class C2UploadedEventHandlerEmailTemplateTest extends EmailTemplateTest {
+    private final C2DocumentBundle c2DocumentBundle = C2DocumentBundle.builder().build();
+    private final String respondentLastName = "Smith";
+    private final String calloutText = "Smith, SACCCCCCCC5676576567";
+    private final String caseUrl = "null/case/" + JURISDICTION + "/" + CASE_TYPE + "/12345#C2Tab";
```

Comment:
Depending on whether my PR https://github.com/hmcts/fpl-ccd-configuration/pull/2063/files is merged first, this will change slightly, but I'll keep an eye on it and let you know if anything needs to change